collated/main/angtianlannus.md

```diff
@@ -0,0 +1,499 @@
+    public String getCurrentViewingAttribute() {
+        return this.currentViewingAttribute;
+    }
+```
+###### \java\seedu\address\model\module\BookedSlot.java
+``` java
+/**
+ * Represents a Booked time slot of a venue
+ */
+public class BookedSlot {
+
+    public final Location location;
+    public final TimeSlot timeSlot;
+
+    public BookedSlot(Location newLocation, TimeSlot newTimeSlot) {
+        this.location = newLocation;
+        this.timeSlot = newTimeSlot;
+    }
+
+    public Location getLocation() {
+        return location;
+    }
+
+    public TimeSlot getTimeSlot() {
+        return timeSlot;
+    }
+
+    /**
+     * Returns true if both booked slot have the same location and time slot(case insensitive)
+     */
+    public boolean isSameStateAs(BookedSlot other) {
+        return this.location.value.toUpperCase().equals(other.getLocation().value.toUpperCase())
+                && this.timeSlot.value.toUpperCase().equals(other.getTimeSlot().value.toUpperCase());
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof BookedSlot // instanceof handles nulls
+                && this.isSameStateAs((BookedSlot) other));
+    }
+
+    @Override
+    public int hashCode() {
+        // use this method for custom fields hashing instead of implementing your own
+        return Objects.hash(location, timeSlot);
+    }
+
+    @Override
+    public String toString() {
+        return location.toString() + "  " + timeSlot.toString();
+    }
+}
+```
+###### \java\seedu\address\model\module\predicates\LessonContainsKeywordsPredicate.java
+``` java
+/**
+ * Tests that a {@code ReadOnlyPerson}'s {@code Phone Number} matches any of the keywords given.
+ */
+public class LessonContainsKeywordsPredicate implements Predicate<ReadOnlyLesson> {
+    private final List<String> keywords;
+    private ArrayList<ReadOnlyLesson> duplicateLessons = new ArrayList<>();
+    private ReadOnlyLesson lesson;
+    private String attribute;
+
+    public LessonContainsKeywordsPredicate(List<String> keywords, ReadOnlyLesson targetLesson, String attribute) {
+
+        this.keywords = keywords;
+        this.lesson = targetLesson;
+        this.attribute = attribute;
+    }
+
+    @Override
+    public boolean test(ReadOnlyLesson lesson) {
+
+        int attribute = -1; // 1 is for module, 2 is for location
+
+        if (lesson.getCode().fullCodeName.equals(this.lesson.getCode().fullCodeName)
+                && this.attribute.equals("module")) {
+            attribute = 1;
+        } else if (lesson.getLocation().value.equals(this.lesson.getLocation().value)
+                && this.attribute.equals("location")) {
+            attribute = 2;
+        } else {
+
+            return false; // not this lesson
+        }
+
+        for (int i = 0; i < keywords.size(); i++) {
+
+            if (attribute == 1) { //search for attribute module code
```

Comment:
Magic number? If you replace the Magic number you can remove the comments as well.